quilt/test/test_import.py

```diff
@@ -0,0 +1,76 @@
+"""
+Tests for magic imports.
+"""
+
+from datetime import datetime
+import os
+
+from quilt.tools import command
+from quilt.tools.const import PACKAGE_DIR_NAME
+from .utils import QuiltTestCase
+
+class ImportTest(QuiltTestCase):
+    def test_imports(self):
+        mydir = os.path.dirname(__file__)
+        build_path = os.path.join(mydir, './build.yml')
+        command.build('foo/package', build_path)
+
+        # Good imports
+
+        import quilt.data.foo.package
+
+        from quilt.data.foo import package
+        from quilt.data.foo.package import dataframes
+        from quilt.data.foo.package import README
+
+        # Good attributes of imported packages
+
+        package.dataframes
+        package.dataframes.csv
+        package.README
+        dataframes.csv
+
+        # Bad attributes of imported packages
+
+        with self.assertRaises(AttributeError):
+            package.foo
+
+        with self.assertRaises(AttributeError):
+            package.dataframes.foo
+
+        with self.assertRaises(AttributeError):
+            package.dataframes.csv.foo
+
+        # Bad imports
+
+        with self.assertRaises(ImportError):
```

Comment:
Do you mean, bad user vs good user with a bad package name? I don't think that's actually possible: the package loader returns either a success or a failure, with no messages.